[Dropbox] Documentation for Advanced sync rules (#1322)
elastic/connectors-python: docs/reference/dropbox.md

Advanced sync rules are not available for this connector in the present version. Currently filtering is controlled via ingest pipelines.

## Advanced Sync Rules

> Advanced Sync Rules are now available in elastic 8.10 version.

Reviewer: ```suggestion
Advanced Sync Rules are available as of Elastic version 8.10.
```

> - Users can add [Dropbox queries](https://www.dropbox.com/developers/documentation/http/documentation#files-search) for syncing the source.

Reviewer: ```suggestion
- Users can add [Dropbox query](https://www.dropbox.com/developers/documentation/http/documentation#files-search) results to their syncs .
```

Correct rephrasing?

>     - Check the search response from [Dropbox API Explorer](https://dropbox.github.io/dropbox-api-v2-explorer/#files_search_v2).

Reviewer: ```suggestion
    - Test search responses using the [Dropbox API Explorer](https://dropbox.github.io/dropbox-api-v2-explorer/#files_search_v2).
```

> - All the data returned by queries will be indexed.

Reviewer: ```suggestion
- All data returned by queries will be indexed.
```

> - You can refer to [DROPBOX.md](../connectors/docs/sync-rules/DROPBOX.md) for the correct format to add sync rules. Any errors encountered, will be presented to the user visually in the sync rules overview.

Reviewer: ```suggestion
- You can refer to [DROPBOX.md](../connectors/docs/sync-rules/DROPBOX.md) for the correct format to add sync rules. Any errors encountered will be reported in the sync rules overview.
```

> **Note:** "path" configuration field will be overridden by the advanced rules.

Reviewer: ```suggestion
**Note:** The "path" configuration field will be overridden by the advanced rules.
```
